<launch>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find planning)/config/rviz_sim.rviz">
  </node>

  <group ns="drone0">
    <!-- rviz中可视化无人机位姿变化 -->
    <node pkg="odom_visualization" name="odom_visualization" type="odom_visualization" output="screen">
      <!-- 将订阅的odom重定向至/drone0/odom -->
      <remap from="~odom" to="odom" />
      <param name="robot_scale" value="1.0"/>
    </node>
    <!-- rviz中可视化平台运动 -->
    <node pkg="odom_visualization" name="odom_visualization_plate" type="odom_visualization_plate" output="screen">
      <remap from="~odom" to="planning/target_odom" />
      <param name="height" value="1.2"/>
      <param name="width" value="1.6"/>
      <param name="platform_r" value="1.0"/>
    </node>

    <node pkg="nodelet" type="nodelet" name="manager" args="manager" output="screen">
      <param name="num_worker_threads" value="16"/>
    </node>

    <node pkg="nodelet" type="nodelet" name="planning" args="load planning/Nodelet manager" output="screen">
      <!-- visuallize用，将发布的/drone0/planning/odom重命名为/drone0/odom -->
      <remap from="~odom" to="odom"/>
      <remap from="~target" to="planning/target_odom"/>
      <!-- 订阅目标与无人机实际odometry -->
      <remap from="~target_odom" to="/odom/remap/car"/>
      <remap from="~uav_odom" to="/odom/remap"/>
      <!-- 发布控制指令 -->
      <remap from="~cmd" to="planning/cmd"/>
      <!-- 外部程序触发规划接口 -->
      <remap from="~triger" to="/landing_triger"/>
      <!-- 控制模块触发信号 -->
      <remap from="~ctrl_triger" to="/traj_start_trigger"/>
      <!-- 未使用话题 -->
      <remap from="~heartbeat" to="heartbeat"/>
      <remap from="~trajectory" to="trajectory"/>
      <remap from="~replanState" to="replanState"/>
      <remap from="~land_triger" to="/land_triger"/>

      <!-- plan rate = plan_hz, cmd pub rate = plan_hz * 10 -->
      <param name="plan_hz" value="5"/>
      <!-- plan_type：0 for sim， 1 for real -->
      <param name="plan_type" value="0"/>
      <param name="visualize_sig" value="true"/>
      <!-- true: run with traj_analyse -->
      <param name="ifanalyse" value="false"/>

      <param name="VehicleMass" value="4.0"/>
      <param name="GravAcc" value="9.8"/>
      <param name="HorizDrag" value="0.70"/>
      <param name="VertDrag" value="0.80"/>
      <param name="ParasDrag" value="0.01"/>
      <param name="SpeedEps" value="0.0001"/>

      <param name="K" value="16"/>
      <param name="vmax" value="1.5"/>
      <!-- <param name="amax" value="0.3"/> -->

      <param name="thrust_max" value="12.8"/>
      <param name="thrust_min" value="6.8"/>
      <!-- <param name="thrust_max" value="12.0"/>
      <param name="thrust_min" value="6.0"/> -->

      <param name="omega_max" value="0.3"/>
      <param name="omega_yaw_max" value="0.17"/>

      <param name="robot_l" value="0.1"/>
      <param name="robot_r" value="0.14"/>
      <param name="platform_r" value="0.3"/>
      <param name="v_plus" value="0.3"/>

      <param name="rhoT" value="10000.0"/>
      <param name="rhoP" value="10000000.0"/>
      <param name="rhoV" value="10000.0"/>
      <!-- <param name="rhoA" value="10000.0"/> -->
      <param name="rhoVt" value="100000.0"/>
      <param name="rhoTf" value="100000.0"/>
      <param name="rhoThrust" value="500000.0"/>
      <param name="rhotailThrust" value="500000.0"/>
      <!-- <param name="rhoThrust" value="10000.0"/> -->
      <param name="rhoOmega" value="100000.0"/>
      <param name="rhoPerchingCollision" value="1000000.0"/>

      <param name="perching_px" value="8.16"/>
      <param name="perching_py" value="8.08"/>
      <param name="perching_pz" value="0.03"/>
      <!-- <param name="perching_pz" value="1.5"/> -->
      <!-- <param name="perching_pz" value="2.0"/> -->
      <param name="perching_vx" value="0.0"/>
      <param name="perching_vy" value="0.0"/>
      <param name="perching_vz" value="0.0"/>

      <!-- axis向量会被归一化 -->
      <!-- <param name="perching_axis_x" value="5.0"/>
      <param name="perching_axis_y" value="0.0"/>
      <param name="perching_axis_z" value="0.0"/> -->
      <param name="perching_axis_x" value="0.0"/>
      <param name="perching_axis_y" value="1.0"/>
      <param name="perching_axis_z" value="0.0"/>
      <!-- <param name="perching_theta" value="-0.0"/> -->
      <!-- <param name="perching_theta" value="0.5"/> -->
      <!-- <param name="perching_theta" value="-0.7854"/> -->
      <!-- <param name="perching_theta" value="-1.5708"/> -->
      <param name="perching_theta" value="0.0"/>

      <param name="replan" value="false"/>
      <param name="pause_debug" value="false"/>

    </node>

  </group>
</launch>